server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailHeader.scala

```diff
@@ -42,11 +48,86 @@ object AddressesHeaderValue extends EmailHeaderValue {
   def from(field: Field): AddressesHeaderValue = AddressesHeaderValue(EmailAddress.from(AddressListFieldImpl.PARSER.parse(field, DecodeMonitor.SILENT).getAddressList))
 }
 
+object GroupedAddressesHeaderValue extends EmailHeaderValue {
+  def from(field: Field): GroupedAddressesHeaderValue = {
+    val addresses: List[Mime4jAddress] =
+      Option(AddressListFieldImpl.PARSER.parse(field, DecodeMonitor.SILENT).getAddressList)
+      .getOrElse(new AddressList())
+      .asScala
+      .toList
+
+    val groups: List[EmailAddressGroup] = addresses
+      .flatMap({
+        case group: Group => Some(group)
+        case _ => None
+      })
+      .map(group => EmailAddressGroup(Some(GroupName(group.getName)), EmailAddress.from(group.getMailboxes)))
+
+    val addressesWithoutGroup: List[EmailAddress] = addresses
+      .flatMap({
+        case mailbox: Mime4jMailbox => Some(mailbox)
+        case _ => None
+      })
+      .map(EmailAddress.from(_))
```

Comment:
you can put the creation of 'groups' and 'addressesWithoutGroup' in the 'else' part of the if